### PR TITLE
[chore] Upgrading Knative Serving to 0.18.3

### PR DESCRIPTION
**Pull request by akirillov**
**What type of PR is this?**
chore

**What this PR does/ why we need it**:
This PR upgrades Knative Serving version to 0.18.3

**Which issue(s) this PR fixes**:
https://jira.d2iq.com/browse/D2IQ-75359

**Special notes for your reviewer**:
The changes are ported from the KNative Serving 0.18.3 release with minor modifications (Go template escaping in `ConfigMap`s): https://github.com/knative/serving/releases/tag/v0.18.3.

The changes were tested manually:
* by upgrading the addon to the new version with the changes from this PR
* running Kaptain workloads that depend on KNative, performing the addon upgrade, and verifying the workloads are not affected

**Does this PR introduce a user-facing change?**:
None

**Release notes**:
```release-note
Knative Serving upgraded to version 0.18.3
```

**Checklist**

* [x] *If a chart is changed, the chart version is correctly incremented.*
* [x] The commit message explains the changes and why are needed.
* [x] The code builds and passes lint/style checks locally.
* [x] The relevant subset of integration tests pass locally.
* [x] The core changes are covered by tests.
* [x] The documentation is updated where needed.

### CODE REVIEW

**Review by joejulian — APPROVED** [review]
lgtm

**Comment by joejulian** [comment]
Is there an expectation that the CRDs should be upgraded when the release is updated? This won't happen automatically. You would need to follow other examples and load the CRDs into a configmap and use a preinstall job to upgrade them. Of course this will only work if the new CRD also has the old storage version (v1alpha1 _and_ v1beta1 for example) or else any custom resources that rely on the old CRD will be deleteed.

**Comment by akirillov** [comment]
thanks for the review, @joejulian. I was about to implement your suggestion looking at the other charts in this repo. Let me implement it and update the PR.

**Review by joejulian — APPROVED** [review]

**Comment by joejulian** [comment]
Loos perfect!

File: staging/knative/charts/serving/templates/configmap-crd.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: knative-serving
+  name: knative-crd
+data:
+  crd.yaml: |-
+{{.Files.Get "crds/crds.yaml" | printf "%s" | indent 4}}
```
Comment (filintod) [inline staging/knative/charts/serving/templates/configmap-crd.yaml:8]: do you need the printf?

**Review by filintod — COMMENTED** [review]
just a little comment

File: staging/knative/charts/serving/templates/configmap-crd.yaml (hunk 1 above)
Comment (akirillov) [reply staging/knative/charts/serving/templates/configmap-crd.yaml:8]: I did it by analogy with Istio (https://github.com/mesosphere/charts/blob/master/staging/istio/templates/configmap-crd.yaml#L8). I guess you need a `printf` to feed its output to `indent` but not sure about that.

Comment (akirillov) [reply staging/knative/charts/serving/templates/configmap-crd.yaml]: just checked that `printf` is not required here. updated the PR

**Review by filintod — APPROVED** [review]
LGTM

**Pull request merged**